android/src/main/java/io/spokestack/RNSpokestack/RNSpokestackModule.java

```diff
@@ -101,47 +99,47 @@ public void initialize(ReadableMap config) throws Exception {
         builder.withoutAutoPlayback();
 
         spokestack = builder.build();
-        onEvent("init", spokestack.getSpeechPipeline().getContext().isActive());
+        WritableMap react_event = Arguments.createMap();
+        react_event.putString("event", "init");
+        sendEvent("init", react_event);
     }
 
     private enum PipelineProfiles {
-        TFWakewordAndroidASR(0),
-        VADTriggerAndroidASR(1),
-        PushToTalkAndroidASR(2),
-        TFWakewordSpokestackASR(3),
-        VADTriggerSpokestackASR(4),
-        PushToTalkSpokestackASR(5);
-
-        private final int profile;
-        PipelineProfiles(int l) {
-            this.profile = l;
+        TFLiteWakewordNativeASR(TFWakewordAndroidASR.class),
+        VADNativeASR(VADTriggerAndroidASR.class),
+        PTTNativeASR(PushToTalkAndroidASR.class),
+        TFLiteWakewordSpokestackASR(TFWakewordSpokestackASR.class),
+        VADSpokestackASR(VADTriggerSpokestackASR.class),
+        PTTSpokestackASR(PushToTalkSpokestackASR.class);
+
+        private final Class<? extends PipelineProfile> profile;
+        PipelineProfiles(Class<? extends PipelineProfile> p) {
+            this.profile = p;
         }
 
-        public int value() {
-            return this.profile;
+        public String value() {
+            return this.profile.getCanonicalName();
```

Comment:
I realize this was my suggestion, and I think it'll be fine given the organization of these things. In my own fooling with this, though, I've noticed that if you ever go to test something like this and use inner classes instead of top-level ones, you might have to switch to `getName` instead of `getCanonicalName`.